lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -103,20 +105,57 @@ def list_jobs(
     )
 
 
-@router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
+@router.get("/{job_spec}")
+def get_job(
+    service: JobServiceDep,
+    job_spec: str,
+    skip: int = 0,
+    limit: int = 100,
+) -> Job | ListingResponse[Job]:
     """Attempts to retrieve merged job data from the Lumigator repository and Ray.
+    If a valid UUID is provided, a single record is returned. If a valid
+    job type (currently `inference` or `evaluation`) is provided, then a list
+    of all jobs of the appropriate type is returned. Any other input results
+    in an error.
 
     NOTE: Lumigator repository data takes precedence over Ray metadata.
     """
-    job = service.get_job(job_id)
-    ray_job = _get_ray_job(job_id)
-
-    # Combine both types of response.
-    x = ray_job.model_dump()  # JobSubmissionResponse
-    y = job.model_dump()  # JobResponse
-    merged = {**x, **y}
-    return Job(**merged)
+    try:
+        job_id = UUID(job_spec)
+        job = service.get_job(job_id)
+        ray_job = _get_ray_job(job_id)
+
+        # Combine both types of response.
+        x = ray_job.model_dump()  # JobSubmissionResponse
+        y = job.model_dump()  # JobResponse
+        merged = {**x, **y}
+        return Job(**merged)
+    except ValueError:
+        # If job_spec is not a UUID, it's assumed to be a
+        if job_spec not in SERVICES_PATHS.values():
+            raise ValueError(f"Unknown job type {job_spec}") from None
+        jobs = service.list_jobs_per_type(job_spec, skip, limit)
+
+        # Get all jobs Ray knows about.
+        ray_jobs = _get_all_ray_jobs()
+
+        results = list[Job]()
+
+        # Merge Ray jobs into the repositories jobs
+        for job in jobs.items:
+            found_job = next(
```

Comment:
Not sure I understand this piece of logic: why are we looking to combine Ray job and lumigator IDs here?

Reply:
This was done in a separate PR already, I'm just following it. The Ray info is bundled with the Lumigator job info so the UI has all the info in one place. Please check https://github.com/mozilla-ai/lumigator/pull/514